### PR TITLE
HTTP/2 Client Example TLS Pipeline Initialization

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
The HTTP/2 client example is not validating the results of ALPN if TLS is enabled.

Modifications:
- Use ApplicationProtocolNegotiationHandler to validate ALPN results.

Result:
Client example validates ALPN results.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@nmittler @normanmaurer - FYI

File: example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java [hunk 1]
```diff
@@ -89,9 +91,22 @@ protected void configureEndOfPipeline(ChannelPipeline pipeline) {
...
-                         connectionHandler);
-        configureEndOfPipeline(pipeline);
+        pipeline.addLast(sslCtx.newHandler(ch.alloc()));
+        // We must wait for the handshake to finish and the protocol to be negotiated before configuring
+        // the HTTP/2 components of the pipeline.
+        pipeline.addLast(new ApplicationProtocolNegotiationHandler("") {
+            @Override
+            protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
+                if (!ApplicationProtocolNames.HTTP_2.equals(protocol)) {
+                    ChannelPipeline p = ctx.pipeline();
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java]: Is the '!' correct here?

Comment (Scottmitch) [inline example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java]: nope ... good catch ... switched the conditional around

**Comment by Scottmitch** [comment]
@normanmaurer - Updated and verified the example runs

File: example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java [hunk 2]
```diff
@@ -89,9 +91,22 @@ protected void configureEndOfPipeline(ChannelPipeline pipeline) {
...
-        pipeline.addLast(sslCtx.newHandler(ch.alloc()),
-                         connectionHandler);
-        configureEndOfPipeline(pipeline);
+        pipeline.addLast(sslCtx.newHandler(ch.alloc()));
+        // We must wait for the handshake to finish and the protocol to be negotiated before configuring
+        // the HTTP/2 components of the pipeline.
+        pipeline.addLast(new ApplicationProtocolNegotiationHandler("") {
+            @Override
+            protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
+                if (ApplicationProtocolNames.HTTP_2.equals(protocol)) {
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java]: Do we need equalsIgnoreCase?

Comment (Scottmitch) [inline example/src/main/java/io/netty/example/http2/helloworld/client/Http2ClientInitializer.java]: No. https://tools.ietf.org/html/rfc7540#section-3.1

**Comment by normanmaurer** [comment]
Ship it

**Comment by Scottmitch** [comment]
Cherry-picked 4.1 (035053b) master (81de3e4)

**Pull request closed without merging**